widgetssdk/src/main/java/com/glia/widgets/chat/domain/UnengagementMessageUseCase.kt

```diff
@@ -0,0 +1,45 @@
+package com.glia.widgets.chat.domain
+
+import com.glia.androidsdk.chat.ChatMessage
+import com.glia.widgets.chat.data.GliaChatRepository
+import com.glia.widgets.core.engagement.GliaEngagementRepository
+import com.glia.widgets.core.engagement.domain.GliaOnEngagementUseCase
+import com.glia.widgets.core.engagement.domain.MapOperatorUseCase
+import com.glia.widgets.core.engagement.domain.model.ChatMessageInternal
+import io.reactivex.Observable
+import java.util.function.Consumer
+
+class UnengagementMessageUseCase(
```

Comment:
I like your name for repository: allMessage, maybe we could name it a similar way, e.g: `AllMessageUseCase` or maybe even `GlobalMessageUseCase`

Reply:
Same in other places

Reply:
It unsubscribes from events after an engagement starts. So, I don't find the `AllMessageUseCase` as a good name for this use case.